### PR TITLE
Feat(TBD-5634)/TBD-5715_correct

### PR DESCRIPTION
**Pull request by mtuhai**
**Please check if the PR fulfills these requirements**

- [X] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [X] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [X] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [X] Docs have been added / updated (for bug fixes / features) ?
- [X] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TBD-5634

**What is the new behavior?**

**BREAKING CHANGE**

CDH5.12.1 implementation

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/CDH5120Distribution.java [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+    }
+
+    @Override
+    public boolean doJavaAPISqoopImportAllTablesSupportExcludeTable() {
+        return true;
+    }
+
+    @Override
+    public boolean doSupportClouderaNavigator() {
+        return true;
```
Comment (pbailly) [inline main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/CDH5120Distribution.java]: I think the cloudera navigator is not integrated to the distribution. If this is the case, please change this boolean to false.

File: main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/CDH5120Distribution.java [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+        return SparkStreamingKafkaVersion.KAFKA_0_10;
+    }
+
+    @Override
+    public short orderingWeight() {
+        return 10;
+    }
+
+    @Override
+    public boolean doImportDynamoDBDependencies() {
```
Comment (pbailly) [inline main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/CDH5120Distribution.java]: DynamoDB is not supported either.

File: main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/CDH5120Distribution.java [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+        return true;
+    }
+
+    @Override
+    public boolean doSupportAzureBlobStorage() {
```
Comment (pbailly) [inline main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/CDH5120Distribution.java]: Azure is not supported either.

File: main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/modulegroup/CDH5120MapReduceModuleGroup.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+public class CDH5120MapReduceModuleGroup {
+
+    public static Set<DistributionModuleGroup> getModuleGroups() {
+        Set<DistributionModuleGroup> hs = new HashSet<>();
+        hs.add(new DistributionModuleGroup(CDH5120Constant.HDFS_MODULE_GROUP.getModuleName()));
+        hs.add(new DistributionModuleGroup(CDH5120Constant.MAPREDUCE_MODULE_GROUP.getModuleName()));
+        hs.add(new DistributionModuleGroup(CDH5120Constant.MAPREDUCE_PARQUET_MODULE_GROUP.getModuleName()));
+        hs.add(new DistributionModuleGroup(CDH5120Constant.MAPREDUCE_PARQUET_MRREQUIRED_MODULE_GROUP.getModuleName(), true, null));
+        ComponentCondition conditionUseNavigator = new SimpleComponentCondition(new BasicExpression(
```
Comment (pbailly) [inline main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/modulegroup/CDH5120MapReduceModuleGroup.java]: Should we add the navigator's dependencies when this is not supported?

File: main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/modulegroup/CDH5120SparkBatchModuleGroup.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            new BasicExpression(SparkBatchConstant.SPARK_LOCAL_MODE_PARAMETER, EqualityOperator.EQ, "false"));
+    
+    public static Set<DistributionModuleGroup> getModuleGroups() {
+        Set<DistributionModuleGroup> hs = new HashSet<>();
+        hs.add(new DistributionModuleGroup(CDH5120Constant.SPARK2_MODULE_GROUP.getModuleName(), false, condition));
+        hs.add(new DistributionModuleGroup(CDH5120Constant.SPARK_MRREQUIRED_MODULE_GROUP.getModuleName(), true, condition));
+        hs.add(new DistributionModuleGroup(CDH5120Constant.HDFS_MODULE_GROUP_SPARK2_2.getModuleName(), false, condition));
+        hs.add(new DistributionModuleGroup(CDH5120Constant.HDFS_MODULE_GROUP_COMMON.getModuleName(), false, condition));
+        hs.add(new DistributionModuleGroup(CDH5120Constant.MAPREDUCE_MODULE_GROUP.getModuleName(), false, condition));
+        ComponentCondition conditionUseNavigator = new SimpleComponentCondition(new BasicExpression(
```
Comment (pbailly) [inline main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/modulegroup/CDH5120SparkBatchModuleGroup.java]: Should we add the navigator's dependencies when this is not supported?

File: main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/CDH5120Distribution.java (hunk 1 above)
Comment (talend-arobu) [reply main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/CDH5120Distribution.java]: I confirm, the Navigator will be integrated in 5.13 and upwards, but not in 5.12.1 and not in 6.5 release

Comment (bchen-talend) [reply main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/CDH5120Distribution.java]: good catch! I'll disable it

File: main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/CDH5120Distribution.java (hunk 2 above)
Comment (bchen-talend) [reply main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/CDH5120Distribution.java]: support it

File: main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/CDH5120Distribution.java (hunk 3 above)
Comment (bchen-talend) [reply main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/CDH5120Distribution.java]: support it

File: main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/modulegroup/CDH5120MapReduceModuleGroup.java (hunk 4 above)
Comment (bchen-talend) [reply main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/modulegroup/CDH5120MapReduceModuleGroup.java]: no, removed

File: main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/modulegroup/CDH5120SparkBatchModuleGroup.java (hunk 5 above)
Comment (bchen-talend) [reply main/plugins/org.talend.hadoop.distribution.cdh5120/src/main/java/org/talend/hadoop/distribution/cdh5120/modulegroup/CDH5120SparkBatchModuleGroup.java]: no, removed

**Pull request merged**